packages/flutter_driver/lib/src/driver/driver.dart

```diff
@@ -155,15 +160,46 @@ class FlutterDriver {
   ///
   /// [isolateReadyTimeout] determines how long after we connect to the VM
   /// service we will wait for the first isolate to become runnable.
+  ///
+  /// [fuchsiaModuleTarget] (optional) If running on a Fuchsia Device, either
+  /// this or the `FUCHSIA_MODULE_TARGET` must be set. Causes
+  /// [isolateNumber] to be ignored/overwritten. This will connect to the
+  /// first Isolate whose name matches.
   static Future<FlutterDriver> connect({
     String dartVmServiceUrl,
     bool printCommunication = false,
     bool logCommunicationToFile = true,
     int isolateNumber,
     Duration isolateReadyTimeout = _kIsolateLoadRunnableTimeout,
+    Pattern fuchsiaModuleTarget,
   }) async {
     dartVmServiceUrl ??= Platform.environment['VM_SERVICE_URL'];
 
+    // If running on a Fuchsia device, connect to the first Isolate whose name
+    // matches FUCHSIA_MODULE_TARGET.
+    if (Platform.isFuchsia) {
```

Comment:
This whole block is all a means of getting the right value for `dartVmServiceUrl`.  Why not just get it before calling `flutterDriver.connect()` and pass it to this method?  This would avoid a Fuchsia-specific dependency in `flutter_driver`.

Reply:
That was the original implementation.  We had a discussion last week involving how to make the `FlutterDriver.connect` function work similarly across platforms and concluded this was the approach to take.

Reply:
We discussed offline - deciding to keep here since there are multiple entry points into flutter driver (Fuchsia in-tree development, internal Google Bazel system, etc.), and it makes sense to keep the code in a unified location.